config.go

```diff
@@ -237,6 +237,11 @@ type Config struct {
 
 	// skipStartup allows NewRaft() to bypass all background work goroutines
 	skipStartup bool
+
+	// NoLegacyTelemetry allow to skip the legacy metrics to avoid duplicates.
+	// legacy metrics are which has `_peer_name` as metric suffix instead as labels.
+	// e.g: raft_replication_heartbeat_peer0
+	NoLegacyTelemetry bool
```

Comment:
Minor thing, but we'd prefer to move this above the private internal-only `skipStartup` option. We typically keep these internal-only details grouped at the end of the struct just to make it clearer at a glance which configs are public.